crates/re_renderer/src/renderer/depth_cloud.rs

```diff
@@ -160,41 +201,50 @@ impl DepthCloudDrawData {
-                colormap: info.colormap as u32,
-                outline_mask_id: info.outline_mask_id.0.unwrap_or_default().into(),
-                end_padding: Default::default(),
-            }),
+            depth_clouds
+                .iter()
+                .map(gpu_data::DepthCloudInfoUBO::from_depth_cloud),
         );
 
         let mut instances = Vec::with_capacity(depth_clouds.len());
         for (depth_cloud, ubo) in depth_clouds.iter().zip(depth_cloud_ubos.into_iter()) {
             let depth_texture = match &depth_cloud.depth_data {
-                // On native, we can use D16 textures without issues, but they aren't supported on
-                // the web (and won't ever be on the WebGL backend, see
-                // https://github.com/gfx-rs/wgpu/issues/3537).
-                //
-                // TODO(cmc): use an RG8 texture and unpack it manually in the shader instead.
-                #[cfg(not(target_arch = "wasm32"))]
                 DepthCloudDepthData::U16(data) => {
-                    create_and_upload_texture(ctx, depth_cloud, data.as_slice(), false)
-                }
-                #[cfg(target_arch = "wasm32")]
-                DepthCloudDepthData::U16(data) => {
-                    use itertools::Itertools as _;
-                    let dataf32 = data
-                        .as_slice()
-                        .iter()
-                        .map(|d| *d as f32 / u16::MAX as f32)
-                        .collect_vec();
-                    create_and_upload_texture(ctx, depth_cloud, dataf32.as_slice(), true)
-                }
-                DepthCloudDepthData::F32(data) => {
-                    create_and_upload_texture(ctx, depth_cloud, data.as_slice(), false)
+                    if cfg!(target_arch = "wasm32") {
+                        // Web: manual normalization because Depth16Unorm textures aren't supported on
+                        // the web (and won't ever be on the WebGL backend, see
+                        // https://github.com/gfx-rs/wgpu/issues/3537).
+                        //
+                        // TODO(cmc): use an RG8 texture and unpack it manually in the shader instead.
+                        use itertools::Itertools as _;
+                        let dataf32 = data
+                            .as_slice()
+                            .iter()
+                            .map(|d| *d as f32 / u16::MAX as f32)
+                            .collect_vec();
+                        create_and_upload_texture(
+                            ctx,
+                            depth_cloud,
+                            dataf32.as_slice(),
+                            wgpu::TextureFormat::R32Float,
+                        )
+                    } else {
+                        // Native: We use Depth16Unorm over R16Unorm because the latter is behind a feature flag,
+                        // and not available on OpenGL backends.
```

Comment:
But didn't we say above that `Depth16Unorm` is not workable on WebGL? We don't actually care about other OpenGL backends.